soroban-env-host/src/host/conversion.rs

```diff
@@ -158,7 +159,6 @@ impl Host {
         durability: ContractDataDurability,
     ) -> Result<Rc<LedgerKey>, HostError> {
         let key_scval = self.from_host_val(k)?;
-        self.check_val_representable_scval(&key_scval)?;
```

Comment:
Why are these being removed?

Reply:
`from_host_val` performs exactly the same check already.